src/action/finder/request.rs

```diff
@@ -0,0 +1,1309 @@
+type DownloadActionOversizedFilePolicy =
+    rrg_proto::file_finder_download_action_options::OversizedFilePolicy;
+type RegexMatchMode =
+    rrg_proto::file_finder_contents_regex_match_condition::Mode;
+type LiteralMatchMode =
+    rrg_proto::file_finder_contents_literal_match_condition::Mode;
+type ActionType = rrg_proto::file_finder_action::Action;
+type ConditionType = rrg_proto::file_finder_condition::Type;
+type XDevMode = rrg_proto::file_finder_args::XDev;
+type PathType = rrg_proto::path_spec::PathType;
+
+#[derive(Debug)]
+pub struct Request {
+    pub paths: Vec<String>,
+    pub path_type: PathType,
+    pub action: Option<Action>,
+    pub conditions: Vec<Condition>,
+    pub process_non_regular_files: bool,
+    pub follow_links: bool,
+    pub xdev_mode: XDevMode,
+}
+
+#[derive(Debug)]
+pub enum Action {
+    Stat(StatActionOptions),
+    Hash(HashActionOptions),
+    Download(DownloadActionOptions),
+}
+
+#[derive(Debug)]
+pub struct StatActionOptions {
+    pub resolve_links: bool,
+    pub collect_ext_attrs: bool,
+}
+
+#[derive(Debug)]
+pub struct HashActionOptions {
+    pub max_size: u64,
+    pub oversized_file_policy: HashActionOversizedFilePolicy,
+    pub collect_ext_attrs: bool,
+}
+
+#[derive(Debug)]
+pub struct DownloadActionOptions {
+    pub max_size: u64,
+    pub oversized_file_policy: DownloadActionOversizedFilePolicy,
+    pub use_external_stores: bool,
+    pub collect_ext_attrs: bool,
+    pub chunk_size: u64,
+}
+
+#[derive(Debug)]
+pub enum Condition {
+    MinModificationTime(std::time::SystemTime),
+    MaxModificationTime(std::time::SystemTime),
+    MinAccessTime(std::time::SystemTime),
+    MaxAccessTime(std::time::SystemTime),
+    MinInodeChangeTime(std::time::SystemTime),
+    MaxInodeChangeTime(std::time::SystemTime),
+    MinSize(u64),
+    MaxSize(u64),
+    ExtFlagsLinuxBitsSet(u32),
+    ExtFlagsLinuxBitsUnset(u32),
+    ExtFlagsOsxBitsSet(u32),
+    ExtFlagsOsxBitsUnset(u32),
+    ContentsRegexMatch(ContentsRegexMatchConditionOptions),
+    ContentsLiteralMatch(ContentsLiteralMatchConditionOptions),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum MatchMode {
+    AllHits,
+    FirstHit,
+}
+
+#[derive(Debug)]
+pub struct ContentsRegexMatchConditionOptions {
+    pub regex: Regex,
+    pub mode: MatchMode,
+    pub bytes_before: u32,
+    pub bytes_after: u32,
+    pub start_offset: u64,
+    pub length: u64,
+}
+
+#[derive(Debug)]
+pub struct ContentsLiteralMatchConditionOptions {
+    pub literal: Vec<u8>,
+    pub mode: MatchMode,
+    pub start_offset: u64,
+    pub length: u64,
+    pub bytes_before: u32,
+    pub bytes_after: u32,
+    pub xor_in_key: u32,
+    pub xor_out_key: u32,
+}
+
+impl TryFrom<rrg_proto::FileFinderAction> for Action {
+    type Error = ParseError;
+
+    fn try_from(
+        proto: rrg_proto::FileFinderAction,
+    ) -> Result<Self, Self::Error> {
+        // FileFinderAction::action_type defines which action will be performed.
```

Comment:
Missing markdown backticks to quote an identifier.

Reply:
Added!